Source/Core/VideoCommon/BPStructs.cpp

```diff
@@ -753,61 +753,396 @@ void LoadBPRegPreprocess(u32 value0)
+      copy_format = "RGBA8";
+      break;
+    case EFBCopyFormat::A8:
+      copy_format = "A8";
+      break;
+    case EFBCopyFormat::R8:
+      copy_format = "R8/I8/Z8H, id 8";
+      break;
+    case EFBCopyFormat::G8:
+      copy_format = "G8/Z8M";
+      break;
+    case EFBCopyFormat::B8:
+      copy_format = "B8/Z8L";
+      break;
+    case EFBCopyFormat::RG8:
+      copy_format = "RG8/Z16R (Note: G and R are reversed)";
+      break;
+    case EFBCopyFormat::GB8:
+      copy_format = "GB8/Z16L";
+      break;
+    case EFBCopyFormat::XFB:
+      // Is this actually a real value, or just an implementation convenience for Dolphin?
+      copy_format = "XFB";
+      break;
+    default:
+      copy_format = fmt::format("Invalid (0x{:x})", static_cast<u32>(copy.tp_realFormat()));
+      break;
+    }
+
+    return format_to(ctx.out(),
+                     "Clamping: {}\n"
+                     "Converting from RGB to YUV: {}\n"
+                     "Target pixel format: {}\n"
+                     "Gamma correction: {}\n"
+                     "Mipmap filter: {}\n"
+                     "Vertical scaling: {}\n"
+                     "Clear: {}\n"
+                     "Frame to field: {}\n"
+                     "Copy to XFB: {}\n"
+                     "Intensity format: {}\n"
+                     "Automatic color conversion: {}",
+                     clamp, no_yes[copy.yuv], copy_format, gamma, no_yes[copy.half_scale],
+                     no_yes[copy.scale_invert], no_yes[copy.clear],
+                     frame_to_field[copy.frame_to_field], no_yes[copy.copy_to_xfb],
+                     no_yes[copy.intensity_fmt], no_yes[copy.auto_conv]);
+  }
+};
+
+template <>
+struct fmt::formatter<TexImage0>
+{
+  constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+  template <typename FormatContext>
+  auto format(const TexImage0& teximg, FormatContext& ctx)
+  {
+    std::string_view texture_format;
+    switch (static_cast<TextureFormat>(teximg.format))
+    {
+    case TextureFormat::I4:
+      texture_format = "I4";
+      break;
+    case TextureFormat::I8:
+      texture_format = "I8";
+      break;
+    case TextureFormat::IA4:
+      texture_format = "IA4";
+      break;
+    case TextureFormat::IA8:
+      texture_format = "IA8";
+      break;
+    case TextureFormat::RGB565:
+      texture_format = "RGB565";
+      break;
+    case TextureFormat::RGB5A3:
+      texture_format = "RGB5A3";
+      break;
+    case TextureFormat::RGBA8:
+      texture_format = "RGBA8";
+      break;
+    case TextureFormat::C4:
+      texture_format = "C4";
+      break;
+    case TextureFormat::C8:
+      texture_format = "C8";
+      break;
+    case TextureFormat::C14X2:
+      texture_format = "C14X2";
+      break;
+    case TextureFormat::CMPR:
+      texture_format = "CMPR";
+      break;
+    case TextureFormat::XFB:
+      // Is this actually a real value, or just an implementation convenience for Dolphin?
```

Comment:
This is not a real format, implementation convenience

Reply:
Removed (along with `EFBCopyFormat::XFB`), but I think it's somewhat odd that `XFB` has the value 0xF allocated to it, when that value is within the normal bounds.  `-1` or `0xffffffff` or something would make it more obvious that it's not something that games would be able to trigger.

Reply:
Honestly, I believe I just took value from phire 's [original code](https://github.com/phire/dolphin/blob/cb84c1e015b0e51bc8cc76e8195c0000dd39550d/Source/Core/VideoCommon/TextureDecoder.h#L65).  I can't say offhand whether it'd be bad to change this to something else, I'd have to go look at the usage, it's been too long!